source/features/repo-age.tsx

```diff
@@ -4,50 +4,108 @@ import cache from 'webext-storage-cache';
+import fetchDom from '../libs/fetch-dom';
 import features from '../libs/features';
-import * as api from '../libs/api';
-import {getRepoGQL, getRepoURL} from '../libs/utils';
+import {getCleanPathname, getRepoURL} from '../libs/utils';
 
 const dateFormatter = new Intl.DateTimeFormat('en-US', {
 	year: 'numeric',
 	month: 'long',
 	day: 'numeric'
 });
 
-const getRepoCreationDate = cache.function(async (): Promise<string> => {
-	const {repository} = await api.v4(`
-		repository(${getRepoGQL()}) {
-			createdAt
-		}
-	`);
+type OldestCommitDetails = {
+	url: string;
+	datetime: string;
+};
 
-	return repository.createdAt;
+const getOldestCommitDetails = cache.function(async (): Promise<OldestCommitDetails | void> => {
+	await elementReady('.commit-tease-sha');
+	const commitsCount = Number(select('li.commits .num')!.textContent!.replace(',', ''));
+	const lastCommitHash = select<HTMLAnchorElement>('.commit-tease-sha')!.href.split('/').pop();
+
+	if (commitsCount === 0) {
+		return;
+	}
+
+	const oldestCommit = await fetchDom(
+		`${getCleanPathname()}/commits?after=${lastCommitHash!}+${commitsCount - 2}`, 'ol:last-child > li.commits-list-item'
+	);
+
+	return {
+		url: select<HTMLAnchorElement>('.commit-title [href*="/commit/"]', oldestCommit)!.href,
+		datetime: select('relative-time', oldestCommit)!.getAttribute('datetime')!
+	};
 }, {
 	cacheKey: () => __featureName__ + ':' + getRepoURL()
 });
 
+const addRepoAgeItem = (element: Element): void => {
+	const license = select('.numbers-summary .octicon-law');
+
+	if (license) {
+		license.closest('li')!.before(element);
+	} else {
+		select('.numbers-summary')!.append(element);
+	}
+};
+
 async function init(): Promise<void> {
-	const date = new Date(await getRepoCreationDate());
+	const placeholder = (
+		<li className="text-gray">
+			{repoIcon()} <span className="text-emphasized js_repo_age_placeholder">Loading...</span>
+		</li>
+	);
+
+	await elementReady('.overall-summary + *');
+
+	addRepoAgeItem(placeholder);
+
+	const details = await getOldestCommitDetails();
+
+	if (!details) {
+		select('.js_repo_age_placeholder', placeholder)!.textContent = '-';
+		placeholder.title = 'Cannot determine repo age from the oldest commit';
+
+		return;
+	}
+
+	// Older cached values are date strings so to handle it for the older
+	// cached values we are checking for the type of the returned values.
+	// This check is supposed to be dropped soon.
+	let oldestCommitUrl;
+	let repoCreationDate;
+
+	if (typeof details === 'object') {
+		oldestCommitUrl = details.url;
+		repoCreationDate = details.datetime;
+	} else {
+		repoCreationDate = details;
+	}
+
+	const date = new Date(repoCreationDate);
 
 	// `twas` could also return `an hour ago` or `just now`
 	const [value, unit] = twas(date.getTime())
 		.replace('just now', '1 second')
 		.replace(/^an?/, '1')
 		.split(' ');
 
-	const element = (
+	const repoAge = <>{repoIcon()} <span className="num text-emphasized">{value}</span> {unit} old</>;
```

Comment:
Instead of regenerating the whole item, you can use the same technique you used to set `Cannot determine repo age from the oldest commit`

In this case, you'll need

```js
select('span', placeholder)!.replaceWith(
	<><span className="num text-emphasized">{value}</span> {unit} old</>
);
```